sdk/sync/watch.go

```diff
@@ -47,57 +44,45 @@ func NewWatcher(runenv *runtime.RunEnv) (w *Watcher, err error) {
 // that point, the caller should consume the error (or nil value) from the
 // returned errCh.
 //
-// The user can cancel the subscription by calling the returned cancelFn. The
-// subscription will die if an internal error occurs, in which case the cancelFn
-// should also be called.
-func (w *Watcher) Subscribe(subtree *Subtree, ch interface{}) (func() error, error) {
+// The user can cancel the subscription by calling the returned cancelFn or by
+// canceling the passed context. The subscription will die if an internal error
+// occurs.
+func (w *Watcher) Subscribe(ctx context.Context, subtree *Subtree, ch interface{}) (context.CancelFunc, error) {
```

Comment:
I'm still returning a cancel function to make it clear that the context _must_ be canceled.

Reply:
Not sure about this pattern... in generally usage of a package/function should be more or less clear. Those functions that require the user to do something else (call `Close()`, or cancel a context), and mostly always documented.

So I am not sure we should add a return value, but you do have a point this is more explicit.

Reply:
What I am trying to say is that we are adding a return value, and we STILL can't know if the user will call it, so we might as well not return it an expect correct usage...

Reply:
SGTM. I'll just go ahead and remove it.